### PR TITLE
Move `propPathOr` to `getPathOr`

### PR DESCRIPTION
**Pull request by evilsoft**
## A Path To Enlightenment (or not)
image

As part of [this issue][issue], this PR addresses the item to move `propPathOr` to the new `getPathOr`. As with all of these moves, `propPathOr` is still available to maintain back-compat until we release `v1.0.0`.

Also fixed an error with the type siggy as part of this. We state we want a foldable, but due to the early exit optimization we added a while back, we need for the path to be specified with an `Array`.

[issue]: https://github.com/evilsoft/crocks/issues/329

### CODE REVIEW

File: docs/src/pages/docs/functions/helpers.md [hunk 1]
```diff
@@ -340,6 +340,59 @@ you provide an `undefined` values for the second, that `Pair` will not be
...
+
+def([ 'foo', 'nan' ], data)
+//=> NaN
+
+def([ 'foo', 'undef' ], data)
+//=> "default"
+
+def([ 'arr', 'length' ], data)
+//=> 2
+```
```
Comment (dalefrancis88) [inline docs/src/pages/docs/functions/helpers.md]: The is no additional changes to the `#### propPathOr` section, would we want to mention there that this is now deprecated and reference the new name?

Comment (dalefrancis88) [reply docs/src/pages/docs/functions/helpers.md]: Also in that example, would we want to switch it to using the new name as well? These are just curiosity questions

File: docs/src/pages/docs/functions/index.md [hunk 2]
```diff
@@ -172,7 +174,6 @@ type: `Pred a` and vice-versa
 [pipes]: helpers.html#pipes
 [prop]: ../crocks/Maybe.html#prop
 [proppath]: ../crocks/Maybe.html#proppath
-[proppathor]: helpers.html#proppathor
```
Comment (dalefrancis88) [inline docs/src/pages/docs/functions/index.md]: So if you take this out, wouldn't you want to remove the section in helpers?

File: src/helpers/getPathOr.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      return def
+    }
+
+    let value = target
+    for(let i = 0; i < keys.length; i++) {
+      const key = keys[i]
+
+      if(!(isString(key) && !isEmpty(key) || isInteger(key))) {
+        throw new TypeError(
+          `${name}: Array of Non-empty Strings or Integers required for second argument`
```
Comment (dalefrancis88) [inline src/helpers/getPathOr.js]: I'm sure that this is pedantic but with strings repeated twice, is the intention to always return the same error? if so would it be worth putting this outside the main function as a small function that takes the name and returns the error?

File: src/helpers/getPathOr.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        `${name}: Array of Non-empty Strings or Integers required for second argument`
+      )
+    }
+
+    if(isNil(target)) {
+      return def
+    }
+
+    let value = target
+    for(let i = 0; i < keys.length; i++) {
```
Comment (dalefrancis88) [inline src/helpers/getPathOr.js]: Is the point of avoiding a recursive function here to not grow the stack?

File: docs/src/pages/docs/functions/helpers.md (hunk 1 above)
Comment (evilsoft) [reply docs/src/pages/docs/functions/helpers.md]: Ohhh great catches. Rebase issue with a stale branch.
Example will be switched and `propPathOr` will be removed from page entirely.
Will update

File: src/helpers/getPathOr.js (hunk 4 above)
Comment (evilsoft) [reply src/helpers/getPathOr.js]: yep and for early exit.

File: src/helpers/getPathOr.js (hunk 3 above)
Comment (evilsoft) [reply src/helpers/getPathOr.js]: I like it, will look into it and update!

Comment (evilsoft) [reply src/helpers/getPathOr.js]: Much better! Great feedback!

File: docs/src/pages/docs/functions/index.md (hunk 2 above)
Comment (evilsoft) [reply docs/src/pages/docs/functions/index.md]: Section removed. Good 👀

File: docs/src/pages/docs/functions/helpers.md (hunk 1 above)
Comment (evilsoft) [reply docs/src/pages/docs/functions/helpers.md]: Section removed.
Great Catch!

**Comment by evilsoft** [comment]
@dalefrancis88 Okay I think I addressed your 🌽cerns, please throw some more 👀 at this if you wanna. And Thanks again!

File: docs/src/pages/docs/functions/index.md [hunk 5]
```diff
@@ -70,6 +70,7 @@ need to account for for the rest of your flow.
 | [`fanout`][fanout] | `(a -> b) -> (a -> c) -> (a -> Pair b c)` | `crocks/Pair/fanout` |
 | [`find`][find] | <code>Foldable f => ((a -> Boolean) &#124; Pred) -> f a -> Maybe a</code> | `crocks/Maybe/find` |
 | [`fromPairs`][frompairs] | `Foldable f => f (Pair String a) -> Object` | `crocks/helpers/fromPairs` |
+| [`getPathOr`][getpathor] | <code>a -> f [ (String &#124; Integer) ] -> b -> a</code> | `crocks/helpers/getPathOr` |
```
Comment (evilsoft) [inline docs/src/pages/docs/functions/index.md]: need to remove that `f`

**Comment by dalefrancis88** [comment]
image

**Pull request merged**